### PR TITLE
FS-161

### PR DESCRIPTION
**Pull request by dhacks541**
Draft pull request for interim feedback 

To do before merging:
- Update with any revised mapping bounds (once confirmed complete, update wiki)
- Rebase off FS-167 once merged
- Update IMU Report to abstract mapping logic (move mapping_bounds to constants.cpp?)
- Test on hardware

Questions:
1. What fields am I missing that we want packed into sensor faults (27-28)?

### CODE REVIEW

File: src/Monitors/NormalReportMonitor.cpp [hunk 1]
```diff
@@ -7,18 +7,106 @@ NormalReportMonitor::NormalReportMonitor(unsigned int offset)
 }
 
 std::queue<uint8_t> NormalReportMonitor::commands_received;
+std::map<uint8_t, std::vector<int>> mapping_bounds = {
```
Comment (ljgreenhill) [inline src/Monitors/NormalReportMonitor.cpp:10]: make sure to limit the max size of the vector

File: src/Monitors/NormalReportMonitor.cpp [hunk 2]
```diff
@@ -7,18 +7,106 @@ NormalReportMonitor::NormalReportMonitor(unsigned int offset)
...
+
+    // Serial.println(sfr::rockblock::normal_report.size());
     sfr::rockblock::normal_report.push_back(constants::rockblock::end_of_normal_downlink_flag1);
-    sfr::rockblock::normal_report.push_back(constants::rockblock::end_of_normal_downlink_flag2);*/
+    sfr::rockblock::normal_report.push_back(constants::rockblock::end_of_normal_downlink_flag2);
+}
+
+uint8_t NormalReportMonitor::serialize(int index, int value)
+{
+    return round((value - mapping_bounds[index][0]) * (255 / (mapping_bounds[index][1] - mapping_bounds[index][0])));
```
Comment (ljgreenhill) [inline src/Monitors/NormalReportMonitor.cpp:111]: Is this supposed to be return round(

(value - mapping_bounds[index][0]) * 

(255 / (mapping_bounds[index][1][1] - mapping_bounds[index][1][0]))

); ?

**Review by ljgreenhill — CHANGES_REQUESTED** [review]

File: src/Monitors/NormalReportMonitor.hpp [hunk 3]
```diff
@@ -12,6 +12,7 @@ class NormalReportMonitor : public TimedControlTask<void>
 
     NormalReportMonitor(unsigned int offset);
     void execute();
+    uint8_t serialize(int index, int value);
```
[the remark this answers is not in the document]
Comment (ljgreenhill) [reply src/Monitors/NormalReportMonitor.hpp:15]: Actually needs to be a float is preserve decimal

**Comment by dhacks541** [comment]
Primary changes in commit d768e14. Packing and serialization of normal report verified. SensorReading debugged to fix pointer issues that were preventing updates to the sensor buffers.

**Pull request closed without merging**